src/core/sys/posix/sys/shm.d

```diff
@@ -156,6 +156,32 @@ else version(NetBSD)
     int   shmdt(in void*);
     int   shmget(key_t, size_t, int);
 }
+else version(DragonFlyBSD)
+{
+    enum SHM_RDONLY     = 0x01000; // 010000
+    enum SHM_RND        = 0x02000; // 020000
+    enum SHMLBA         = 1 << 12; // PAGE_SIZE = (1<<PAGE_SHIFT)
+
+    alias c_ulong   shmatt_t;
+
+    struct shmid_ds
+    {
+         ipc_perm    shm_perm;
+         int         shm_segsz;
+         pid_t       shm_lpid;
+         pid_t       shm_cpid;
+         short       shm_nattch;
+         time_t      shm_atime;
+         time_t      shm_dtime;
+         time_t      shm_ctime;
+         void *      __shm_internal;
```

Comment:
DStyle: `void*` (also do you have any resource where it's `__shm_internal`. I did a quick search, but could only find `shm_internal` like it's for FreeBSD.

Reply:
On Dfly, /usr/include/sys/shm.h looks like:
```
struct shmid_ds {
    struct ipc_perm shm_perm;   /* operation permission structure */
    size_t          shm_segsz;  /* size of segment in bytes */
    pid_t           shm_lpid;   /* process ID of last shared memory op */
    pid_t           shm_cpid;   /* process ID of creator */
    shmatt_t    shm_nattch; /* number of current attaches */
    time_t          shm_atime;  /* time of last shmat() */
    time_t          shm_dtime;  /* time of last shmdt() */
    time_t          shm_ctime;  /* time of last change by shmctl() */
    void           *shm_internal;   /* sysv stupidity */
};
```
Have not looked at the use or implementation of this header file. I did have to gloss over the comment for this pointer (Logically it would be the opaque pointer to an internal representation/expansion for this struct,). But i thought size might matter (doesn't it always).

Reply:
Thanks.
I would suggest to keep the `shm_internal` name then. You could set it to private though?

> But i thought size might matter 

Yes.